micro_sam/training/semantic_sam_trainer.py

```diff
@@ -56,11 +85,13 @@ def _train_epoch_impl(self, progress, forward_context, backprop):
 
             backprop(net_loss)
 
+            self._iteration += 1
+
             if self.logger is not None:
                 lr = [pm["lr"] for pm in self.optimizer.param_groups][0]
-                self.logger.log_train(self._iteration, net_loss, lr, x, y, masks, log_gradients=True)
+                predictions = torch.softmax(masks, dim=1)
```

Comment:
I don't think it's  necessary to do the softmax here; we would then do it twice (see comment below)